src/components/view/pages/main/filters/filtersView.ts

```diff
@@ -0,0 +1,263 @@
+            });
+        }
+
+        const stockFilter = document.querySelector('.filters-item .stock') as HTMLElement | null;
+        if (stockFilter) {
+            this.initializeSlider(stockFilter, (start, end) => {
+                Router.removeUrlParamKey('page');
+                Router.setUrlParam('minStock', start)
+                Router.setUrlParam('maxStock', end)
+            });
+        }
+
+        const colors = document.querySelectorAll<HTMLElement>('.colors__color')
+        colors.forEach(c => c.addEventListener('click', () => {
+            Router.removeUrlParamKey('page');
+            if (c.classList.contains('is-selected')) {
+                Router.removeUrlParamValue('colors', c.dataset.color || '');
+            } else {
+                Router.addUrlParamValue('colors', c.dataset.color || '');
+            }
+        }))
+
+        const collections = document.querySelectorAll<HTMLElement>('.collection__year')
+        collections.forEach(c => c.addEventListener('click', () => {
+            Router.removeUrlParamKey('page');
+            if (c.classList.contains('is-selected')) {
+                Router.removeUrlParamValue('collections', c.dataset.collection || '');
+            } else {
+                Router.addUrlParamValue('collections', c.dataset.collection || '');
+            }
+        }))
+
+        const categories = document.querySelectorAll<HTMLInputElement>('.category__checkbox')
+        categories.forEach(c => c.addEventListener('change', () => {
+            Router.removeUrlParamKey('page');
+            if (!c.checked) {
+                Router.removeUrlParamValue('categories', c.dataset.categories || '');
+            } else {
+                Router.addUrlParamValue('categories', c.dataset.categories || '');
+            }
+        }))
+
+        const closeButton = document.querySelector<HTMLElement>('.filters__close-btn')
+        closeButton?.addEventListener('click', () => {
+            Router.removeUrlParamKey('showFilters')
+        })
+    }
+
+    private initializeSlider(container: HTMLElement, onChange: (min: string, max:string) => void, maxDecimals: number = 0) {
+        const slider = container.nextElementSibling?.querySelector('.slider') as HTMLElement | null;
+        const startInput = container.querySelector('.box-start') as HTMLInputElement | null;
+        const endInput = container.querySelector('.box-end') as HTMLInputElement | null;
+
+        if (!slider || !startInput || !endInput) {
+            return
+        }
+
+        const minValue = parseFloat(slider.dataset.min ?? '')
+        const maxValue = parseFloat(slider.dataset.max ?? '')
+        let startMin = minValue
+        let startMax = maxValue
+
+        if (Number(slider.dataset.min_selected)) {
+            startMin = Number(slider.dataset.min_selected)
+        }
+        if (Number(slider.dataset.max_selected)) {
+            startMax = Number(slider.dataset.max_selected)
+        }
```

Comment:
Try to avoid `let` and code duplication
const minSelected = +slider.dataset.min_selected
const startMin = minSelected ? minSelected : minValue